plugins/database/mysql/connection_producer.go

```diff
@@ -140,14 +172,33 @@ func (c *mySQLConnectionProducer) Connection(ctx context.Context) (interface{},
 		// If the ping was unsuccessful, close it and ignore errors as we'll be
 		// reestablishing anyways
 		c.db.Close()
+
+		// if IAM authentication was enabled
+		// ensure open dialer is also closed
+		if c.AuthType == connutil.AuthTypeGCPIAM {
+			if c.cloudDialerCleanup != nil {
```

Comment:
I don't see where `c.cloudDialerCleanup` is being set

Reply:
I think we are ignoring the cleanup func returned from `registerDriverMySQL`?

Reply:
Thanks for the catch! Updated in latest commits